pallets/unicks/src/lib.rs

```diff
@@ -0,0 +1,443 @@
+		///   deposit release
+		/// # </weight>
+		#[pallet::weight(0)]
+		pub fn blacklist(origin: OriginFor<T>, unick: UnickInput<T>) -> DispatchResult {
+			T::BlacklistOrigin::ensure_origin(origin)?;
+
+			let unick = UnickOf::<T>::try_from(unick.into_inner()).map_err(DispatchError::from)?;
+
+			// No failure beyond this point
+
+			// Unregister (including returning the deposit) only if the unick was assigned
+			// to someone.
+			if Self::check_blacklisting_preconditions(&unick)? {
+				Self::unregister_unick_unsafe(&unick);
+			}
+
+			Self::blacklist_unick_unsafe(&unick);
+			Self::deposit_event(Event::<T>::UnickBlacklisted { unick });
+
+			Ok(())
+		}
+
+		/// Unblacklist a unick.
+		///
+		/// Make a unick claimable again.
+		///
+		/// The origin must be the blacklist origin.
+		///
+		/// Emits `UnickUnblacklisted` if the operation is carried out
+		/// successfully.
+		///
+		/// # <weight>
+		/// Weight: O(1)
+		/// - Reads: Blacklist storage entry + origin check
+		/// - Writes: Blacklist storage entry deposit release
+		/// # </weight>
+		#[pallet::weight(0)]
+		pub fn unblacklist(origin: OriginFor<T>, unick: UnickInput<T>) -> DispatchResult {
+			T::BlacklistOrigin::ensure_origin(origin)?;
+
+			let unick = UnickOf::<T>::try_from(unick.into_inner()).map_err(DispatchError::from)?;
+
+			Self::check_unblacklisting_preconditions(&unick)?;
+
+			// No failure beyond this point
+
+			Self::unblacklist_unick_unsafe(&unick);
+			Self::deposit_event(Event::<T>::UnickUnblacklisted { unick });
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Verify that the claiming preconditions are verified. Specifically:
+		/// - The unick input data can be decoded as a valid unick
+		/// - The unick does not already exist
+		/// - The owner does not already own a unick
+		/// - The unick has not been blacklisted
+		/// - The tx submitter has enough funds to pay the deposit
+		fn check_claiming_preconditions(
+			unick: UnickInput<T>,
+			owner: &UnickOwnerOf<T>,
+			deposit_payer: &AccountIdOf<T>,
+		) -> Result<UnickOf<T>, DispatchError> {
+			let unick = UnickOf::<T>::try_from(unick.into_inner()).map_err(DispatchError::from)?;
+
+			ensure!(!Unicks::<T>::contains_key(&owner), Error::<T>::UnickAlreadyClaimed);
+			ensure!(!Owner::<T>::contains_key(&unick), Error::<T>::OwnerAlreadyExisting);
+			ensure!(!Blacklist::<T>::contains_key(&unick), Error::<T>::UnickBlacklisted);
+
+			ensure!(
+				<T::Currency as ReservableCurrency<AccountIdOf<T>>>::can_reserve(
+					deposit_payer,
+					<T as Config>::Deposit::get()
+				),
+				Error::<T>::InsufficientFunds
+			);
+
+			Ok(unick)
+		}
+
+		/// Assign a unick to the provided owner reserving the deposit from the
+		/// provided account. This function must be called after
+		/// `check_claiming_preconditions` as it does not verify all the
+		/// preconditions again.
+		fn register_unick_unsafe(unick: UnickOf<T>, owner: UnickOwnerOf<T>, deposit_payer: AccountIdOf<T>) {
```

Comment:
Yeah but the main thing is that, for instance, that the `blacklist` extrinsic calls internally the `unregister_unick_unsafe` function, and we don't want to perform all the checks again. Since the unsafe functions provide basic CRUD functionalities, they might be needed in different places with different requirements, so putting the checks inside those functions I don't think it's ideal. Maybe I should add a safe alternative that is then called as-is from within each extrinsic?